### ISSUE TITLE
anchors should not be considered as links

### ISSUE THREAD

**Issue by oupala** [davidxmoody/metalsmith-broken-link-checker#5]
The `a` tag can be a link or an anchor.

`<a name="an-anchor">anchor</a>` is an anchor, while `<a href="#an-anchor>link to the anchor</a>` is a link to the previously defined anchor.

Currently, the plugin considers anchors as links.

This causes a buggy output that was a bit hard to debug :

```
Warning: Link is broken: href: "undefined", text: "", in file: my-path\index.html
Warning: Link is broken: href: "undefined", text: "", in file: my-path\index.html
```

Anchors are not links and should not be considered as such.

**Comment by davidxmoody** [comment]
Completely valid point, thanks for bringing it up. I've added an `allowAnchors` option (defaults to true) to treat anchors as valid. Just published it as version 0.1.5. 

I know you raised a few other issues a while ago which I've been procrastinating on. I'm still thinking about them. I might add the ability to provide a custom handler function for handling all links. It could accept arguments specifying whether the link is broken or absolute and so on. Then you could log it to a file or do something else with it. What do you think?

**Comment by oupala** [comment]
Thanks David for this bugfix !

I can now see that the default behaviour does not warn any more on anchors.

However, I don't get the point of the allowAnchors parameter. Anchors are never links, they can't be invalid links !

Could you explain me a bit what you had in mind about this option ? What could it be used for ?

Your proposal for the other issues seems great.

**Comment by davidxmoody** [comment]
I guess I just included the option because it was easy to do so. In retrospect I see what you mean, it doesn't really make much sense to ever disallow them. I will probably remove that option in the future.
